andela/ah-django-unchained
# #162948854 Implement  sharing of  articles

#### What does this PR do?
Allows users to share articles

#### Description of Task to be completed?
Authenticated users should be able to share articles on their social media accounts (facebook and twitter) or via email

#### How should this be manually tested?

- Clone the repo locally
- cd ah-django-unchained
- git checkout ft-share-articles-162948854
- get started with the app as stated in the README section
- to add an article ensure you obtain a token from logging in or signing up if not signed up
- Using the `POST api/articles/` endpoint, have the following JSON object. Note that image uploads are optional.
```
{
    "title": "My story",
    "description": "I am very happy",
    "body": "A story about me",
    "tagList": ["reactjs", "angularjs", "dragons"]

}
```
- To share the article on Facebook
```
GET api/articles/<slug>/share/facebook
```

- To share the article via Email

```
GET api/articles/<slug>/share/email/
```
After navigating to that url pass in a valid email as a query parameter such that the url now conforms to this pattern `GET api/articles/<slug>/share/email/?email=example@mail.com`

- To share the article on Twitter
```
GET api/articles/<slug>/share/twitter
```

#### Relevant Pivotal Tracker Stories

[#162948854](https://www.pivotaltracker.com/story/show/162948854)
